system-addon/lib/HighlightsFeed.jsm

```diff
@@ -49,6 +52,18 @@ this.HighlightsFeed = class HighlightsFeed {
 
   uninit() {
     SectionsManager.disableSection(SECTION_ID);
+    PageThumbs.removeExpirationFilter(this);
+  }
+
+  filterForThumbnailExpiration(callback) {
+    const sectionIndex = SectionsManager.sections.get(SECTION_ID).order;
+    const {initialized, rows} = this.store.getState().Sections[sectionIndex];
```

Comment:
Looks like it's possible for this line to throw if the section doesn't exist in the state. Seems unlikely but if it were to break, it looks like thumbnail expiration filter isn't smart enough to recover and waits forever.